BREAKING.md

```diff
@@ -17,6 +17,21 @@ It's important to communicate breaking changes to our stakeholders. To write a g
 - Avoid using code formatting in the title (it's fine to use in the body).
 - To explain the benefit of your change, use the [What's New](https://fluidframework.com/docs/updates/v1.0.0/) section on FluidFramework.com.
 
+# 2.1.0
+## 2.1.0 Breaking changes
+- [Remove `type` field from `ShareLinkInfoType`](#Remove-type-field-from-ShareLinkInfoType)
+- [Remove `ShareLinkTypes` interface](#Remove-ShareLinkTypes-interface)
+- [Remove `enableShareLinkWithCreate` from `HostStoragePolicy`](#Remove-enableShareLinkWithCreate-from-HostStoragePolicy)
+### Remove `type` field from `ShareLinkInfoType`
+This field has been deprecated and will be removed in a future breaking change.
+
+### Remove `ShareLinkTypes` field from `ShareLinkInfoType`
+This interface has been deprecated and will be removed in a future breaking change. Functions impacted are:
+- `createOdspCreateContainerRequest`
+
+
+### Remove `enableShareLinkWithCreate` from `HostStoragePolicy`
+This field has been deprecated and will be removed in a future breaking change.
```

Comment:
@tylerbutler / @jatgarg 
If the idea is to make the three items I list here as deprecated for now, and remove them in the next major release, how should the Breaking.md declaration look here?

I have made changes here based on what I saw in the file here, but I am not sure if this is the correct approach.

Reply:
Understood. The change is backwards compatible and I am just announcing certain things as deprecated. Should I then change the above to: ## 3.0.0 Breaking changes?

Reply:
@tylerbutler

Reply:
This is a very confusing area at the moment. ☹️  I think that since the removal will be in next, the announcement should be in the _Upcoming 2.0.0 Changes_ section. @scottn12 Is that right? Or can we start using the new 2.0 version scheme in BREAKING.md? I think that would clarify things.

Reply:
> This is a very confusing area at the moment. ☹️ I think that since the removal will be in next, the announcement should be in the _Upcoming 2.0.0 Changes_ section. @scottn12 Is that right? Or can we start using the new 2.0 version scheme in BREAKING.md? I think that would clarify things.

Yes Tyler, that's correct. A good rule of thumb is that minor releases should never have breaking changes. As Tyler said these should all be listed under "Upcoming Changes". I actually recently wrote up a short wiki page to better explain these guidelines [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes). Feel free to take a look and let me know if anything there is confusing (or any other feedback). I'd love to hear if it helps!

Reply:
Thanks for the guideline link, it was quite helpful. Suggestion to add this link to the `Does this introduce a breaking change?`  section of the PR description. (If not already there) @scottn12